src/index.ts

```diff
@@ -427,82 +630,150 @@ export default class OneSignal {
+     * @param  {string} key
+     * @returns Promise<string>
+     */
+    static getTriggerValueForKey(key: string): Promise<string | void> {
         // must return a promise
         if (!isObjectNonNull(RNOneSignal)) return Promise.resolve();
         return RNOneSignal.getTriggerValueForKey(key);
     }
 
-    static pauseInAppMessages(pause) {
+    /**
+     * Pause & unpause In-App Messages
+     * @param  {boolean} pause
+     * @returns void
+     */
+    static pauseInAppMessages(pause: boolean): void {
         if (!isObjectNonNull(RNOneSignal)) return;
         RNOneSignal.pauseInAppMessages(pause);
     }
 
     /* O U T C O M E S */
 
-    static sendOutcome(name, handler=function(){}) {
+    /**
+     * Increases the "Count" of this Outcome by 1 and will be counted each time sent.
+     * @param  {string} name
+     * @param  {(event:OutcomeEvent)=>void} handler
+     * @returns void
+     */
+    static sendOutcom(name: string, handler?: (event: OutcomeEvent) => void): void {
         if (!isObjectNonNull(RNOneSignal)) return;
         RNOneSignal.sendOutcome(name, handler);
     }
 
-    static sendUniqueOutcome(name, handler=function(){}) {
+    /**
+     * Increases "Count" by 1 only once. This can only be attributed to a single notification.
+     * @param  {string} name
+     * @param  {(event:OutcomeEvent)=>void} handler
+     * @returns void
+     */
+    static sendUniqueOutcome(name: string, handler?: (event: OutcomeEvent) => void): void {
         if (!isObjectNonNull(RNOneSignal)) return;
         RNOneSignal.sendUniqueOutcome(name, handler);
     }
 
-    static sendOutcomeWithValue(name, value, handler=function(){}) {
+    /**
+     * Increases the "Count" of this Outcome by 1 and the "Sum" by the value. Will be counted each time sent.
+     * If the method is called outside of an attribution window, it will be unattributed until a new session occurs.
+     * @param  {string} name
+     * @param  {string|number} value
+     * @param  {(event:OutcomeEvent)=>void} handler
+     * @returns void
+     */
+    static sendOutcomeWithValue(name: string, value: string | number, handler?: (event: OutcomeEvent) => void): void {
         if (!isObjectNonNull(RNOneSignal)) return;
         RNOneSignal.sendOutcomeWithValue(name, Number(value), handler);
     }
 
     /* P R I V A C Y  C O N S E N T */
 
-    static userProvidedPrivacyConsent() {
+    /**
+     * Did the user provide privacy consent for GDPR purposes.
+     * @returns Promise<boolean>
+     */
+    static userProvidedPrivacyConsent(): Promise<boolean | void> {
         if (!isObjectNonNull(RNOneSignal)) return Promise.resolve();
 
         //returns a promise
         return RNOneSignal.userProvidedPrivacyConsent();
     }
 
-    static requiresUserPrivacyConsent() {
+    /**
+     * True if the application requires user privacy consent, false otherwise
+     * @returns Promise<boolean>
+     */
+    static requiresUserPrivacyConsent(): Promise<boolean | void> {
```

Comment:
Should not return void. True or False (void ~> default to false)

Reply:
Done, resolve to `false`